engine/structures.hpp

```diff
@@ -23,16 +23,16 @@ enum class PointerType : uint8_t {
   SkiplistNode = 3,
   // Point to a dram Skiplist object
   Skiplist = 4,
-  // Point to a UnorderedCollection object on DRAM
-  UnorderedCollection = 5,
-  // Point to a element of UnorderedCollection
-  UnorderedCollectionElement = 6,
-  // Point to a dram List object
+  // Hash
+  HashList = 5,
+  // Element in Hash
+  HashElem = 6,
```

Comment:
we had hold snapshot when iterating hash. HashElem could be removed.

Reply:
Please explain this in more details.